include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type
```

Comment:
```
         if(isnan(in[ii]))
```
I'm surprised that this has never failed on any other platform -- is it really conceivable that everyone else implements `isnan(int)`?

It also appears that `std::enable_if<std::is_integral<T>::value, bool>::type` is already supported by Visual Studio 2010. Are there other old compilers that might cause problems here?